src/App.js

```diff
@@ -1,23 +1,337 @@
-import logo from './logo.svg';
-import './App.css';
+import './index.css';
+import { useState, useEffect } from 'react';
+import axios from 'axios';
+import { TailSpin } from 'react-loader-spinner';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faBookReader, faHeart as heart } from '@fortawesome/free-solid-svg-icons';
+import { faHeart as farHeart } from '@fortawesome/free-regular-svg-icons';
+
+
+// components
+import Book from './components/Book';
+import Form from './components/Form';
+import Modal from './components/Modal';
+import BookCart from './components/BookCart';
+import firebaseProject from './firebaseSetup';
+import { getDatabase, ref, onValue, push } from 'firebase/database';
+import FilterBooks from './components/FilterBooks';
+import Footer from './components/Footer';
+
 
 function App() {
+  const [books, setBooks] = useState([]);
+  const [searchTerm, setSearchTerm] = useState('');
+  const [isHidden, setIsHidden] = useState(false);
```

Comment:
Nice, good technique. I'm not sure how constructive my comments will be so I'll litter humor here and there along with actual thoughts. Maybe. Maybe not, I think this process is the exercise moreso. But everything looks good so far.